### PR TITLE
Create `Io` interface and refactor fs-specific code into `NativeIo`

### PR DESCRIPTION
**Pull request by bjaspan**
This refactoring supports the creation of a web versions of this extension, though it does not yet include one.

### CODE REVIEW

File: src/nativeAdapter.ts [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+	!stats.isDirectory() ||
+	stats.isSymbolicLink() ||
+	stats.mode & S_IWOTH ||
+	// On Windows, uid < 0, so we don't worry about it for simplicity
+	(info.uid >= 0 && stats.uid !== info.uid)
+	) {
+      throw new Error(`Refusing to proceed because of invalid communication dir ${communicationDirPath}`);
+    }
+
+    return new Promise(() => {});
```
Comment (bjaspan) [inline src/nativeAdapter.ts]: I'm not sure about this line. I need to do some more reading...

Comment (bjaspan) [reply src/nativeAdapter.ts]: Removed.

**Comment by pokey** [comment]
ok sorry for the long delay. I pushed [a couple changes](https://github.com/pokey/command-server/pull/19/files/2df87600fa76eb9525efb1e0389eca06b6447733..6f2785a06bfd14beb220e44459ed45443811b79a). Look ok? I'm going to run this for a couple days just to be sure, as the command server unfortunately doesn't have any tests, and is fundamental to cursorless and pretty much all Talon-based VSCode usage 😅

**Comment by bjaspan** [comment]
The changes look fine to me.

**Review by pokey — APPROVED** [review]
Ok been working fine for a couple days so I think we can ship it

**Pull request merged**